src/main/host/syscall/handler/mman.rs

```diff
@@ -95,13 +95,22 @@ impl SyscallHandler {
         let addr: usize = addr.try_into().unwrap();
         let addr = ForeignPtr::<()>::from(addr).cast::<u8>();
 
-        // check for truncated flag bits (use u32 instead of i32 to prevent sign extension when
-        // casting from signed to unsigned)
-        if prot as u32 as u64 != prot {
-            warn_once_then_trace!("Ignoring truncated prot flags from mprotect: {prot}");
-        }
-
-        let prot = prot as i32;
+        let Some(prot) = ProtFlags::from_bits(prot) else {
+            for flag in ProtFlags::from_bits_retain(prot)
+                .difference(ProtFlags::all())
+                .iter()
+            {
```

Comment:
I don't think the iter is needed here. The value that `iter()` is called on will only be unknown bit values, and `iter()` returns all unknown bit values together as a single value.

> Each yielded flags value will correspond to a defined named flag. Any unknown bits will be yielded together as a final flags value.

Since this iterator will only ever return a single value of all the unknown bits, I think it would be simpler just to log `ProtFlags::from_bits_retain(prot).difference(ProtFlags::all())` directly.

Reply:
Done. Breaking it into the individual bits would be preferable, but I'll leave that yak alone for now.